cli/admin/call.go

```diff
@@ -0,0 +1,193 @@
+package admin
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/apex/log"
+	"github.com/tarantool/cartridge-cli/cli/project"
+
+	"github.com/spf13/pflag"
+
+	"github.com/tarantool/cartridge-cli/cli/common"
+	"github.com/tarantool/cartridge-cli/cli/templates"
+)
+
+const (
+	argTypeString = "string"
+	argTypeNumber = "number"
+	argTypeBool   = "boolean"
+)
+
+type FuncCallArg struct {
+	Name string
+	Type string
+
+	StringValue string
+	NumberValue int
+	BoolValue   bool
+
+	Changed bool
+}
+
+func adminFuncCall(conn net.Conn, funcName string, flagSet *pflag.FlagSet, args []string) error {
+	funcCallArgs, err := getFuncCallArgs(conn, funcName, flagSet, args)
+	if err != nil {
+		return fmt.Errorf("Failed to parse function call args: %s", err)
+	}
+
+	argsSerialized, err := serializeArgs(funcCallArgs)
+	if err != nil {
+		return fmt.Errorf("Failed to serialize function args: %s", err)
+	}
+
+	callFuncBody, err := templates.GetTemplatedStr(&adminCallFuncBodyTmpl, map[string]string{
+		"AdminCallFuncName": adminCallFuncName,
+		"FuncName":          funcName,
+		"Args":              argsSerialized,
+	})
+
+	callResRaw, err := common.EvalTarantoolConn(conn, callFuncBody)
+	if err != nil {
+		return fmt.Errorf("Failed to call %q: %s", funcName, err)
+	}
+
+	printCallRes(callResRaw)
+
+	return nil
+}
+
+func getFuncCallArgs(conn net.Conn, funcName string, flagSet *pflag.FlagSet, args []string) ([]FuncCallArg, error) {
+	helpResRawMap, err := getFuncHelpRawMap(funcName, conn)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get function %q signature: %s", funcName, err)
+	}
+
+	argsSpec, err := getArgsSpec(helpResRawMap)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get %q arguments spec: %s", funcName, err)
+	}
+
+	conflictingFlagNames := getConflictingFlagNames(argsSpec, flagSet)
+	if len(conflictingFlagNames) > 0 {
+		return nil, fmt.Errorf(
+			"Function has arguments with names that conflicts with `cartridge admin` flags: %s",
+			strings.Join(conflictingFlagNames, ", "),
+		)
+	}
+
+	funcCallArgs := make([]FuncCallArg, len(argsSpec))
+
+	i := 0
```

Comment:
`i` -> `arg_num`?

Reply:
I sure that `i` is a good name for such cases, it's just an index.